x/gov/keeper/deposit.go

```diff
@@ -266,3 +267,24 @@ func (keeper Keeper) validateInitialDeposit(ctx context.Context, initialDeposit
 	}
 	return nil
 }
+
+func (keeper Keeper) validateDepositDenom(ctx context.Context, params v1.Params, depositAmount sdk.Coins) error {
+	denoms := []string{}
+	acceptedDenoms := make(map[string]bool, len(params.MinDeposit))
+	for _, coin := range params.MinDeposit {
+		acceptedDenoms[coin.Denom] = true
+		denoms = append(denoms, coin.Denom)
+	}
+
+	if len(depositAmount) > len(params.MinDeposit) {
```

Comment:
this doesnt seem right, if the array length is longer than the other then it fails? is this to say that you cant deposit more than the listed amount of denoms, but if amount of accepted denoms is 3 and you submit two of different denoms this would pass? maybe its not needed?

Reply:
This means you need to deposit the same amount of denoms that it is specified in gov params.
This is just a fast check before looping through depositAmount.

This does enforce that you always need to deposit each time all accepted denoms. I can remove this check.

Reply:
https://github.com/cosmos/cosmos-sdk/pull/18189/commits/11f1eafc6afec36e56eac9dd4875646c992affe6